src/components/VmsPageHeader/index.js

```diff
@@ -15,25 +16,42 @@ import OverlayTooltip from '../OverlayTooltip'
 /**
  * Main application header on top of the page
  */
-const VmsPageHeader = ({ page, onRefresh }) => {
-  const idPrefix = `pageheader`
-  return (
-    <Header>
-      <div className='collapse navbar-collapse'>
-        <ul className='nav navbar-nav navbar-right navbar-iconic'>
-          <li>
-            <OverlayTooltip id={`${idPrefix}-tooltip`} tooltip={msg.refresh()} placement='bottom'>
-              <a href='#' className='nav-item-iconic' onClick={hrefWithoutHistory(() => onRefresh(page))} id={`${idPrefix}-refresh`}>
-                <i className='fa fa-refresh' />
-              </a>
-            </OverlayTooltip>
-          </li>
-          <UserMenu />
-          <VmUserMessages />
-        </ul>
-      </div>
-    </Header>
-  )
+class VmsPageHeader extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      show: false,
+    }
+
+    this.handleToggle = this.handleToggle.bind(this)
+  }
+
+  handleToggle () {
+    this.setState((prevState) => ({ show: !prevState.show }))
+  }
```

Comment:
I suggest using React hooks for this kind of thing going forward.  Those allow us to keep function components and still manage state as needed.

For example: https://gerrit.ovirt.org/gitweb?p=ovirt-engine-ui-extensions.git;a=blob;f=src/components/modals/vm/VmMigrateModal.js;h=5b089b4fa212e40b5d5a2c0cf51c3f1195ce1885;hb=HEAD#l19